### PR TITLE
Finish Coll.flatMap implementation

### LINKED ISSUE #423: Error thrown: "java.lang.ArrayStoreException: java.lang.Byte"
Currently ignore test "flatMap" (in CollectionOperationsSpecification) throwing an error.

The script is 
OUTPUTS.flatMap({ (out: Box) => out.propositionBytes })(0) == 0.toByte

the error is:
Caused by: scalan.Base$StagingException: Cannot invoke method public abstract special.collection.Coll special.collection.Coll.flatMap(scala.Function1,scalan.RType) on object Coll(...)
...
Caused by: java.lang.ArrayStoreException: java.lang.Byte
	at scala.collection.mutable.ArrayBuilder$ofRef.$plus$eq(ArrayBuilder.scala:91)
	at scala.collection.mutable.ArrayBuilder$ofRef.$plus$eq(ArrayBuilder.scala:60)
	at scala.collection.generic.Growable.$anonfun$$plus$plus$eq$1(Growable.scala:62)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)

### PR DESCRIPTION
**Pull request by greenhat**
Close #423

### CODE REVIEW

File: src/main/scala/sigmastate/eval/CostingRules.scala [hunk 1]
```diff
@@ -548,18 +548,20 @@ trait CostingRules extends SigmaLibrary { IR: RuntimeCosting =>
     }
 
     def flatMap[B](fC: RCosted[T => Coll[B]]): RCostedColl[B] = {
+      val sizeF = asCostedFunc[T, Coll[B]](fC).sliceSize
```
Comment (aslesarenko) [inline src/main/scala/sigmastate/eval/CostingRules.scala]: @greenhat ok, but it is better done after the pattern below is checked.

Comment (greenhat) [reply src/main/scala/sigmastate/eval/CostingRules.scala]: Thanks! Fixed.

**Comment by aslesarenko** [comment]
@greenhat please also add a case to SigmaDslTests

**Comment by greenhat** [comment]
@aslesarenko I did not like how `CostingSpec` test came up, so I rewrite `flatMap` costing. Check it out, please.

**Pull request merged**